activejob/lib/active_job/enqueuing.rb

```diff
@@ -9,9 +9,9 @@ module Enqueuing
 
     # Includes the +perform_later+ method for job initialization.
     module ClassMethods
-      # Push a job onto the queue. The arguments must be legal JSON types
-      # (+string+, +int+, +float+, +nil+, +true+, +false+, +hash+ or +array+) or
-      # GlobalID::Identification instances. Arbitrary Ruby objects
+      # Push a job onto the queue. The arguments must be either String,
+      # Integer, Float, NilClass, TrueClass, FalseClass, BigDecimal, Hash,
+      # Array or GlobalID::Identification instances. Arbitrary Ruby objects
```

Comment:
Since #30941, Active Job have some arguments serializers by default.
How about reflecting the Supported types for arguments in the guide to the doc?

https://github.com/rails/rails/blame/9bd5fc84ccaa59bd9ca8e81145e5ce15fbac75d2/guides/source/active_job_basics.md#L342-L356

Reply:
It definitely makes sense @kamipo . Thanks also for showing me that part of the code, I didn't know that before ❤️ 

I have updated the doc and also the commit message to reference that PR.